src/mds/MDSMap.cc

```diff
@@ -153,11 +153,9 @@ void MDSMap::dump(Formatter *f) const
   for (set<mds_rank_t>::const_iterator p = in.begin(); p != in.end(); ++p)
     f->dump_int("mds", *p);
   f->close_section();
-  f->open_object_section("up");
+  f->open_array_section("up");
   for (map<mds_rank_t,mds_gid_t>::const_iterator p = up.begin(); p != up.end(); ++p) {
-    char s[14];
-    sprintf(s, "mds_%d", int(p->first));
-    f->dump_int(s, p->second);
+    f->dump_int("mds", p->first);
```

Comment:
@batrick 

to double confirm with you, if follow you suggestions, the output of `up` section would generate output as below?

rank -> gid

```
  "up": [
    {
      "mds_0": 1234
    },
    {
      "mds_1": 2334
    },
    {
      "mds_2": 3455
    }
  ]
```

or this one?


```
  "up": [
    {
      0: 1234
    },
    {
      1: 2334
    },
    {
      2: 3455
    }
  ]
```

Reply:
Hmm, in that case I think I prefer what you have now (an array of ranks for `up` and an array for `info`). The user of the JSON output will have to do some post-processing to make it easier to use programatically. It would have been nice to avoid but JSON is rather dumb.

This PR still needs to fix the volume client, any tests that may be processing `up` as an object and `info` as an object, and update `PendingReleaseNotes` to note the output format change.